components/odh-notebook-controller/main.go

```diff
@@ -58,16 +59,20 @@ func init() {
 	utilruntime.Must(routev1.AddToScheme(scheme))
 	utilruntime.Must(configv1.AddToScheme(scheme))
 	utilruntime.Must(oauthv1.AddToScheme(scheme))
+	utilruntime.Must(imagev1.AddToScheme(scheme))
 
 	//+kubebuilder:scaffold:scheme
 }
 
 func getControllerNamespace() (string, error) {
 	// Try to get the namespace from the service account secret
+	// or from the K8S_NAMESPACE environment variable in local
 	if data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
 		if ns := string(data); len(ns) > 0 {
 			return ns, nil
 		}
+	} else if ns := os.Getenv("K8S_NAMESPACE"); len(ns) > 0 {
```

Comment:
This is not necessary to add. @sven has already added a constant that holds the namespace in tests. It has ridiculously long name. I'll look it up when I'm back in office.

Reply:
https://github.com/opendatahub-io/kubeflow/blob/72ac94883bf9b7944c3bfee11f638e05393fc395/components/odh-notebook-controller/controllers/suite_test.go#L77

Reply:
Local run fails, as there is no namespace set at that path
to avoid that included this env var
https://github.com/opendatahub-io/kubeflow/blob/72ac94883bf9b7944c3bfee11f638e05393fc395/components/odh-notebook-controller/Makefile#L111

Reply:
Cool, so you're fixing a bug I did not even know we're having!